### PR TITLE
feat: support 1s candles

### PR DESCRIPTION
**Pull request by rise0chen**
Notice: test will failed, because of https://github.com/Nouzan/indicator/pull/28

### CODE REVIEW

**Comment by Nouzan** [comment]
https://github.com/Nouzan/indicator/pull/28 has been merged and published with `v0.4.0-beta.0`. You can rebase and try again.

**Comment by rise0chen** [comment]
ok

File: exc-okx/src/utils/period.rs [hunk 1]
```diff
@@ -140,3 +145,49 @@ pub fn bar_to_period(bar: &str) -> Option<Period> {
...
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    fn test_bar(bar: &str) {
+        let period = bar_to_period(bar).unwrap_or_else(|| panic!("{}", bar));
+        let out_bar = period_to_bar(&period).unwrap_or_else(|| panic!("{} - {}", bar, period));
```
Comment (Nouzan) [inline exc-okx/src/utils/period.rs:155]: Maybe better use `.expect(&format!("{bar}"));`.

File: exc-okx/src/utils/period.rs [hunk 2]
```diff
@@ -140,3 +145,49 @@ pub fn bar_to_period(bar: &str) -> Option<Period> {
...
+        assert_eq!(bar, out_bar);
+    }
+    fn test_period(period: Period) {
+        let bar = period_to_bar(&period).unwrap_or_else(|| panic!("{}", period));
+        let out_period = bar_to_period(bar).unwrap_or_else(|| panic!("{} - {}", period, bar));
```
Comment (Nouzan) [inline exc-okx/src/utils/period.rs:160]: Same as above.

**Review by Nouzan — CHANGES_REQUESTED** [review]
Thank you for the improvements. Once the style issues are resolved, I will be happy to accept this PR.

**Comment by rise0chen** [comment]
It is suggested by [clippy](https://rust-lang.github.io/rust-clippy/master/index.html#expect_fun_call)

**Comment by Nouzan** [comment]
> It is suggested by [clippy](https://rust-lang.github.io/rust-clippy/master/index.html#expect_fun_call)

Oh, I see.

**Pull request merged**